listener.go

```diff
@@ -191,8 +191,19 @@ func ensureFilterRule(
 	subName string,
 	filterName string,
 	filter servicebus.FilterDescriber) error {
+	rules, err := sm.ListRules(ctx, subName)
+	if err != nil {
+		return err
+	}
+	for _, rule := range rules {
```

Comment:
that prevents you from updating a rule once it's there.
you should check that the content of the rule is also matching

Reply:
good call. if the rule exists and the contents are the same, then i will return early. otherwise if the rule exists but the contents are different then i will allow for an update of the rule.